aioshelly/__init__.py

```diff
@@ -162,10 +162,14 @@ async def create(
         ip_or_options: Union[str, ConnectionOptions],
     ):
         """Device creation."""
+        loop = asyncio.get_running_loop()
         if isinstance(ip_or_options, str):
-            options = ConnectionOptions(gethostbyname(ip_or_options))
+            ip_addr = await loop.run_in_executor(None, gethostbyname, ip_or_options)
```

Comment:
Why not just add this resolving after the if…else check and do it only if it's not an IP?